bot/cogs/duck_pond.py

```diff
@@ -0,0 +1,213 @@
+import logging
+from typing import List, Optional, Union
+
+import discord
+from discord import Color, Embed, Member, Message, PartialEmoji, RawReactionActionEvent, Reaction, User, errors
+from discord.ext.commands import Bot, Cog
+
+from bot import constants
+from bot.utils.messages import send_attachments
+
+log = logging.getLogger(__name__)
+
+
+class DuckPond(Cog):
+    """Relays messages to #duck-pond whenever a certain number of duck reactions have been achieved."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+        self.log = log
+        self.webhook_id = constants.Webhooks.duck_pond
+        self.bot.loop.create_task(self.fetch_webhook())
+
+    async def fetch_webhook(self) -> None:
+        """Fetches the webhook object, so we can post to it."""
+        await self.bot.wait_until_ready()
+
+        try:
+            self.webhook = await self.bot.fetch_webhook(self.webhook_id)
+        except discord.HTTPException:
+            self.log.exception(f"Failed to fetch webhook with id `{self.webhook_id}`")
+
+    @staticmethod
+    def is_staff(member: Union[User, Member]) -> bool:
+        """Check if a specific member or user is staff."""
+        if hasattr(member, "roles"):
+            for role in member.roles:
+                if role.id in constants.STAFF_ROLES:
+                    return True
+        return False
+
+    @staticmethod
+    def has_green_checkmark(message: Optional[Message] = None, reaction_list: Optional[List[Reaction]] = None) -> bool:
+        """Check if the message has a green checkmark reaction."""
+        assert message or reaction_list, "You can either pass message or reactions, but not both, or neither."
+
+        if message:
+            reactions = message.reactions
+        else:
+            reactions = reaction_list
+
+        for reaction in reactions:
+            if isinstance(reaction.emoji, str):
+                if reaction.emoji == "✅":
+                    return True
+            elif isinstance(reaction.emoji, PartialEmoji):
```

Comment:
I don't think supporting `PartialEmoji` is necessary. In practice, only message objects retrieved with `fetch_message()` are used:

> This model (PartialEmoji) will be given in two scenarios:
>
>    * “Raw” data events such as on_raw_reaction_add()
>    * Custom emoji that the bot cannot see from e.g. Message.reactions

Neither apply since the message object is neither from a raw event nor is the green checkmark a custom emoji. I feel like we shouldn't bother since this is function is meant for internal use rather than as a general-purpose utility.

Reply:
This, as well as the other `reaction_list` options for the two helpers comes from an `on_reaction_clear` handler that I have removed because it wasn't really working, just FYI. They did serve a purpose, but they don't anymore. Good spot.